viewing_party/party.py

```diff
@@ -1,23 +1,276 @@
+        return "Intrigue"
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+# finds the movies the user has watched AND their friends have not
+def get_unique_watched(user_data):
+    friend_movies = set()
+    user_movies = set()
+    unique_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_movies.add(movie["title"])
+    for movie in user_data["watched"]:
+        user_movies.add(movie["title"])
+    unique_titles = user_movies - friend_movies
+    for movie in user_data["watched"]:
+        if movie["title"] in unique_titles:
+            unique_watched.append(movie)
+    return unique_watched
+
+# finds the movies the user's friends have watched that they have not
+def get_friends_unique_watched(user_data):
+    friend_movies = set()
+    user_movies = set()
+    unique_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_movies.add(movie["title"])
+    for movie in user_data["watched"]:
+        user_movies.add(movie["title"])
+    unique_titles = friend_movies - user_movies
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] in unique_titles:
+                if movie not in unique_watched:
+                    unique_watched.append(movie)
+    return unique_watched
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+# finds the movies the user hasn't watched, their friends have, AND they have access to (through their streaming service(s))
+def get_available_recs(user_data):
+    recs = []
+    friends_movies = get_friends_unique_watched(user_data)
+    for movie in friends_movies:
+        if movie["host"] in user_data["subscriptions"]:
+            recs.append(movie)
+    return recs
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+# CONTEXT: my first attempt at finding the mode of the genre
+# def get_new_rec_by_genre(user_data):
+#     all_recs = get_available_recs(user_data)
+#     genre_dict = {}
+#     for movie in user_data["watched"]:
+#         for genre in movie["genre"]:
+#             if genre not in genre_dict:
+#                 genre_dict[genre] = 1
+#             else:
+#                 genre_dict[genre] += 1
+#     fav_genre = max(genre_dict)
+#         counter = 0
+#         # for genre in genre_dict:
+#         #     if genre_dict[genre] > 
+
+# gets recommendations for the user where:
+# they haven't watched it, their friends have, they have access to it, AND it's their most watched genre
+def get_new_rec_by_genre(user_data):
+    rec_by_genre = []
+    all_recs = get_available_recs(user_data)

```

Comment:
This is the function call that necessitated modifying the test data above. This is actually not the function you want to use. From Wave 3, if you use `get_friends_unique_watched` and `get_most_watched_genre` from Wave 2, you have all you need to accomplish the task. That would look something like: 
```python
    most_freq_genre = get_most_watched_genre(user_data)
    unique_friend_movies = get_friends_unique_watched(user_data)

    recs = []

    for movie in unique_friend_movies:
        if movie['genre'] == most_freq_genre:
            recs.append(movie)

    return recs
```